frontend/src/__tests__/cypress/cypress/utils/oc_commands/baseCommands.ts

```diff
@@ -51,3 +51,22 @@ export const patchOpenShiftResource = (
     return result;
   });
 };
+
+/**
+ * Wait for a specific pod to become ready across all namespaces.
+ *
+ * @param podNameContains A substring to partially match against the pod name (e.g., "jupyter-nb").
+ * @returns A Cypress chainable that waits for the pod to be ready.
+ */
+export const waitForPodReady = (podNameContains: string): Cypress.Chainable<CommandLineResult> => {
+  const ocCommand = `oc get pods -A --no-headers | awk '$2 ~ /^${podNameContains}/ {print $1, $2}' | xargs -tn2 oc wait --for=condition=Ready pod --timeout=400s -n`;
```

Comment:
Can you add 2 more args to the function ?

- `timeout` - with default of 10 seconds) - It should be String, since timeout could be `10s`, `10m`, `1h`.
- `namespace` - with default of `--all-namespaces / -A`. When passing a namespace it should be `-n <namespace>` instead of the `-A`

Reply:
refactored